File: gitdot-web/packages/gitdot-api/src/endpoint/user/list-user-repositories-starred.ts
import { z } from "zod";
import { page, UserRepositoryResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const ListUserStarredRepositoriesRequest = z.object({
  cursor: z.string().optional(),
  limit: z.number().int().positive().optional(),
});
export type ListUserStarredRepositoriesRequest = z.infer<
  typeof ListUserStarredRepositoriesRequest
>;

export const ListUserStarredRepositoriesResponse = page(UserRepositoryResource);
export type ListUserStarredRepositoriesResponse = z.infer<
  typeof ListUserStarredRepositoriesResponse
>;

export const ListUserStarredRepositories = {
  path: "/user/{user_name}/repositories-starred",
  method: "GET",
  request: ListUserStarredRepositoriesRequest,
  response: ListUserStarredRepositoriesResponse,
} as const satisfies Endpoint;
export type ListUserStarredRepositories = typeof ListUserStarredRepositories;

File: gitdot-api/src/endpoint/user/list_user_repositories_starred.rs
use serde::{Deserialize, Serialize};

use crate::{
    endpoint::Endpoint,
    resource::{common::Page, user::UserRepositoryResource},
};

pub struct ListUserStarredRepositories;

impl Endpoint for ListUserStarredRepositories {
    const PATH: &'static str = "/user/{user_name}/repositories-starred";
    const METHOD: http::Method = http::Method::GET;

    type Request = ListUserStarredRepositoriesRequest;
    type Response = ListUserStarredRepositoriesResponse;
}

#[derive(ApiRequest, Debug, Default, Serialize, Deserialize)]
pub struct ListUserStarredRepositoriesRequest {
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub cursor: Option<String>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub limit: Option<u32>,
}

pub type ListUserStarredRepositoriesResponse = Page<UserRepositoryResource>;
